Cache price data

TristanFecteau opened the pull request:
We need to merge cachePriceData into main, because this is our new way to store fmp data.

Be careful, both of these branches have seen quite a few changes since they split.

TristanFecteau committed:
can now cache price data upon initializing an FmpDataSource and acces the stocks values from the cached stock list

Still lot of work to do, currently only returns close value, can only have one data source, plus a lot of testing to do with different Resolutions, but it works

space333k committed:
storage de donnee completer et skipage des jour ferier ajouter, Il reste a clean up et rendre le nombre de donner enregistrer flexible

space333k committed:
fix bug de limit de donnée par request

TristanFecteau committed:
Merge branch 'main' into cachePriceData

TristanFecteau committed:
fixed a missing return misallocated with auto-merge

SamiMammouche commented on src/bullets/data_source/data_source_fmp.py:
```diff
@@ -12,40 +14,96 @@ def __init__(self, token: str, resolution: Resolution):
         self.resolution = resolution
+        self.stocks = {}
+
+    def store_price_points(self, symbol: str, start_date: datetime.date, end_date: datetime.date = None,
```
Are people coding their strategies supposed to call this method? If not, you can mark it as private by adding an underscore at the beginning (`_store_price_points`)

SamiMammouche commented on src/bullets/data_source/data_source_fmp.py:
```diff
@@ -12,40 +14,96 @@ def __init__(self, token: str, resolution: Resolution):
+
+        self.stocks[symbol] = stock
+
+    def get_market_open_close(self):
```
The logic here is good, I'm just wondering if we should rely on the FMP endpoint. Since it doesn't provide enough data in the past, we'll still need to find another way to fix the holiday issue.

Maybe we should take this logic out of this pull request and re-evaluate what we need to do in the holiday issue.

SamiMammouche commented on src/bullets/portfolio/transaction.py:
```diff
@@ -1,3 +1,5 @@
+
```
You can remove these two blank lines

SamiMammouche commented on src/bullets/portfolio/holding.py:
```diff
@@ -1,3 +1,5 @@
+
```
You can remove these two blank lines

TristanFecteau committed:
removed holiday logic

-Removed a few empty lines
-Removed an unused function in fmp
-Made storePricePointsPrivate
-Had to change the getPrice logic as the previous version relied on fmp always having the data asked, hence not taking into account the fact that there are holidays and such, wich is important to add because we removed the holiday logic in fmp
-Changed transaction count in test_startegy since we now do one more transaction because it is on a holiday (we used to skip that day)

TristanFecteau committed:
cleanup
